### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by andriy-shymkiv**

### CODE REVIEW

File: src/sortStudents.ts [hunk 1]
```diff
@@ -1,16 +1,73 @@
...
+          ? pupil1.age - pupil2.age
+          : pupil2.age - pupil1.age;
+      });
+
+    case SortType.Married:
+      return studentsCopy.sort((pupil1: Student, pupil2: Student) => {
+        return order === 'asc'
+          ? +pupil1.married - +pupil2.married
+          : +pupil2.married - +pupil1.married;
+      });
```
Comment (karyna-salnykova) [inline src/sortStudents.ts:61]: I think you can use here
`case 1:
case 2:
  //code`

to make your code shorter and unite statements that do the same

File: src/sortStudents.ts [hunk 2]
```diff
@@ -1,16 +1,73 @@
...
+          ? pupil1.name.localeCompare(pupil2.name)
+          : pupil2.name.localeCompare(pupil1.name);
+      });
+
+    case SortType.Surname:
+      return studentsCopy.sort((pupil1: Student, pupil2: Student) => {
+        return order === 'asc'
+          ? pupil1.surname.localeCompare(pupil2.surname)
+          : pupil2.surname.localeCompare(pupil1.surname);
+      });
```
Comment (karyna-salnykova) [inline src/sortStudents.ts:47]: The same advice as in SortType.Age and SortType.Married part

File: src/sortStudents.ts [hunk 3]
```diff
@@ -1,16 +1,73 @@
...
+  grades: Array<number>;
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
```
Comment (karyna-salnykova) [inline src/sortStudents.ts:15]: I think you should write which value of Student should be in every part. Don't know how to explain what I mean, but it's like:

`Name = 'name'`

**Review by karyna-salnykova — CHANGES_REQUESTED** [review]

File: src/sortStudents.ts [hunk 4]
```diff
@@ -1,16 +1,61 @@
...
+    .reduce((acc: number, mark: number) => acc + mark, 0) / gradesArray.length;
+}
+
+export function sortStudents(
+  students: Array<Student>,
+  sortBy: SortType,
+  order: SortOrder,
+)
+
+  : Array<Student> {
```
Comment (illiarerun) [inline src/sortStudents.ts:31]: formatting
```suggestion
)  : Array<Student> {
```

**Review by illiarerun — APPROVED** [review]

**Pull request closed without merging**